### PR TITLE
Set potentially unset variables to null. 

### PR DESCRIPTION
**Pull request by SlaterByte**
This satisfies issue #181 and has been tested with both bash and zsh with 'set -ue' and with 'set -o pipeline' with bash.

### CODE REVIEW

**Comment by ahmadia** [comment]
I just reviewed this.  It seems that in every case you prefer the `:=` form of parameter expansion (sets the variable) as opposed to `:` (tests the variable, but doesn't modify it).  Don't we want the latter in most of these cases?

File: bin/activate [hunk 1]
```diff
@@ -3,7 +3,9 @@
 # Ensure that this script is sourced, not executed
 # Also note that errors are ignored as `activate foo` doesn't generate a bad
 # value for $0 which would cause errors.
-if [[ -n $BASH_VERSION ]] && [[ "$(basename "$0" 2> /dev/null)" == "activate" ]]; then
+# Also note, variables that could potentially be unset should have parameter
+# expansion set them to null. ex: ${VARIABLE:=''}
+if [[ -n ${BASH_VERSION:=''} ]] && [[ "$(basename "$0" 2> /dev/null)" == "activate" ]]; then
```
Comment (ahmadia) [inline bin/activate]: There's no reason to set BASH_VERSION in this test.

File: bin/activate [hunk 2]
```diff
@@ -12,9 +14,9 @@ instead of 'activate envname'.
 fi
 
 # Determine the directory containing this script
-if [[ -n $BASH_VERSION ]]; then
+if [[ -n ${BASH_VERSION:=''} ]]; then
```
Comment (ahmadia) [inline bin/activate]: Same here.  I think you want: `${BASH_VERSION:''}` instead.

**Comment by ahmadia** [comment]
I like the idea of this.  I think the preferred form that should be safe in both Bash and Zsh is `${VAR:-''}`.

**Comment by ahmadia** [comment]
In this situation `${VAR-''}` should be morally equivalent.

**Comment by ahmadia** [comment]
@SlaterByte - it would be great if we could get this fix in.  Let me know if you'd like to try a second pass at this.

**Comment by kalefranz** [comment]
This file has now moved to github.com/conda/conda

**Pull request closed without merging**